### PR TITLE
pushing the repo to this branch with prediction stage script update and code restructuring

### PR DESCRIPTION
**Pull request by NSR9**
# Proposed Changes

Before
The repo version on the branch sprint_dev is squashed and merged on to master untiil now. That version has the Image Fetcher and Pre-processing stages of the pipeline. 

After
Branched onto this new branch called "chay/addingPredictionStage" sourcing from "sprint_dev" which has following changes:
* Created a config.ini file.
* Built my_utils folder which has image fetcher and pre-processing script. 
* Modified the plv_module_pipeline.py
* Modularized the code for image_fetcher and preprocessing scripts

# Jira Stories

- [LAB-293](https://oregonstate-innovationlab.atlassian.net/browse/LAB-293?atlOrigin=eyJpIjoiYzEwOTRmNTE2MTI5NDZjMTk3MWI0NGZiNGUwMWVjZTQiLCJwIjoiaiJ9)

# Checklist

[1] This has been tested
[1] Self-documenting code
[1] Code is linted

# Instructions to review
Here are the steps:
* Clone the repository.
* Change directory using "cd ms_final_project"
* Install the required packages by running "pip install -r requirements.txt"
* then run "python3 plv_module1_pipeline.py"

### CODE REVIEW

File: my_utils/image_fetcher.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    capture.open(best.url)
+
+    success,fetched_image = capture.read()
+
+    current_date = datetime.now()
+    str_date = current_date.strftime('%Y-%m-%dT%H:%M:%S.%f%z')
+
+    
+    # initializing size of string
+    N = 7
```
Comment (pavan3008) [inline my_utils/image_fetcher.py]: Can you please rename it with proper context?

File: plv_module1_pipeleine.py [hunk 2]
```diff
@@ -6,66 +6,99 @@
...
-  
-# Path
-dir_path = os.path.join(parent_dir, directory)
+def process2_pipeline(MODEL, MONGO, AWS_CREDENTIALS, AWS_ENV,PREPROCESSING_CONFIG):
+    # IMAGE FETCHER SCRIPT
+    url = "https://www.youtube.com/watch?v=e9LYewJGQlk"
+
+    fetched_image, filename_string, str_date = image_fetcher(url)
+
+    # etting the path for saving the fetched and preprocessed images.
```
Comment (pavan3008) [inline plv_module1_pipeleine.py]: I assume it is meant to be "setting"

File: plv_module1_pipeleine.py [hunk 3]
```diff
@@ -6,66 +6,99 @@
...
+    preprocess_image_dir_path = PREPROCESSING_CONFIG["preprocessed-image-dir-save-path"]
+    
+    # Sending the image to preprocessing script 
+    pre_processed_image = pre_process(fetched_image, preprocess_image_dir_path, filename_string) 
   
+    # Setting the directory path to fetch the image for detection 
+    source = PREPROCESSING_CONFIG["preprocessed-image-dir-save-path"] + filename_string
+
+
+    # DETECTION/ PREDICT part of the process pipeline
```
Comment (pavan3008) [inline plv_module1_pipeleine.py]: Detection/Prediction *

**Review by pavan3008 — CHANGES_REQUESTED** [review]
Code looks fine. I've added some suggestions. Please go through it.

**Comment by NSR9** [comment]
Updated the repo with suggested changes

**Review by pavan3008 — APPROVED** [review]

**Pull request merged**